### PR TITLE
Update Overview and Dependencies pages

### PR DESCRIPTION
**Pull request by adasiewiczr**
### 🎯 Goal

Update `Overview` and `Dependencies` pages

### 🎨 UI Changes

| Overview | Dependencies |
| --- | --- |
| localhost_3000_chat_docs_sdk_android_|localhost_3000_chat_docs_sdk_android_basics_dependencies_|

### 🧪 Testing

Run `npx stream-chat-docusaurus -i -s` and check pages

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_

### CODE REVIEW

File: docusaurus/docs/Android/01-basics/01-overview.mdx [hunk 1]
```diff
@@ -29,27 +30,42 @@ The client library is a low-level wrapper around the Stream Chat API. It lets yo
 
 It also contains all the basic model objects you'll interact with when using Stream Chat, such as `User`, `Channel`, or `Message`.
 
-### Offline Support
+### State
+
+The state library is a plugin that gives you the possibility to reactively observe data through `StateFlows`.
+Moreover, the plugin adds a sync feature, that for example is responsible for resending messages, reactions, etc, when the user performed actions without an Internet connection.
```
Comment (MarinTolic) [inline docusaurus/docs/Android/01-basics/01-overview.mdx:36]:
```suggestion
It adds a sync feature which handles re-sending messages, reactions, and syncing other actions the user had performed while offline.
```

File: docusaurus/docs/Android/01-basics/03-dependencies.mdx [hunk 2]
```diff
@@ -79,6 +89,18 @@ dependencies {
 
 Adding the Compose UI Components library as a dependency will automatically include the client and offline libraries as well.
 
+### Markdown support
+
+We ship additional artifact for Markdown support that can be used together with UI Components library:
```
Comment (MarinTolic) [inline docusaurus/docs/Android/01-basics/03-dependencies.mdx:94]:
```suggestion
We ship an additional artifact for Markdown support which can be used together with the UI Components library:
```

File: docusaurus/docs/Android/01-basics/01-overview.mdx [hunk 3]
```diff
@@ -29,27 +30,42 @@ The client library is a low-level wrapper around the Stream Chat API. It lets yo
...
+
+:::note
+The state plugin doesn't add a persistence layer.<br/> Consider using it together with `Offline Plugin`.
+:::
 
-The offline support is a plugin that adds offline caching capabilities using a local database. For example, it allows you to send messages or add reactions while you're offline. When the user comes back online, the library will automatically recover lost events and retry sending messages.
+<!-- TODO update link once State documentation is ready -->
+This plugin exposes easy-to-use `StateFlow` objects for reading channels, messages, reads, typing, members, watchers and more. See [State](../02-client/06-guides/04-offline-support.mdx) for more details.
```
Comment (MarinTolic) [inline docusaurus/docs/Android/01-basics/01-overview.mdx:43]: Maybe we should list  `ChannelState` members descriptively here by what they do, such as:

```
This plugin exposes easy-to-use `StateFlow` objects containing information about the channel such as the messages inside it, a list of users currently typing, a list of all channel members and more.
```

The idea is that user reading this is a new user and might not know the nomenclature well.
This is just a suggestion, feel free to reject it if you do not like it.

File: docusaurus/docs/Android/01-basics/01-overview.mdx [hunk 4]
```diff
@@ -29,27 +30,42 @@ The client library is a low-level wrapper around the Stream Chat API. It lets yo
...
+
+If you want to build custom UI for Stream Chat, you can build it on top of the state library.
```
Comment (MarinTolic) [inline docusaurus/docs/Android/01-basics/01-overview.mdx:45]:
```suggestion
If you want to build a custom UI for Stream Chat, you can build it on top of the state library.
```

File: docusaurus/docs/Android/01-basics/01-overview.mdx [hunk 5]
```diff
@@ -29,27 +30,42 @@ The client library is a low-level wrapper around the Stream Chat API. It lets yo
...
+* Decreases initial load times, especially when the network connection is poor.
+* Enables the user to see cached channels and messages while offline.
+* Enables the user to perform actions such as sending messages and reactions while offline.
 
 ### UI Components
 
 The UI Components library includes pre-built Android Views to easily load and display data from the Stream Chat API. These include a Channel List and a Message List, a Message Composer View, and more. See the UI Components [Getting Started](../03-ui/01-overview.mdx) page for more details.
 
-This library is built on top of the offline support library, and offers the quickest integration of Stream Chat into an Android application. It also has a variety of [theming](../03-ui/02-general-customization/01-theming.mdx) options to make it fit your app's needs.
+This library is built on top of the state library, and offers the quickest integration of Stream Chat into an Android application. It also has a variety of [theming](../03-ui/02-general-customization/01-theming.mdx) options to make it fit your app's needs.
```
Comment (MarinTolic) [inline docusaurus/docs/Android/01-basics/01-overview.mdx:60]:
```suggestion
This library is built on top of the state library, and offers the quickest integration of Stream Chat into an Android application. It also offers a variety of [theming](../03-ui/02-general-customization/01-theming.mdx) options to make it fit your app's needs.
```

**Review by MarinTolic — APPROVED** [review]

**Review by skydoves — APPROVED** [review]
Looks good to me!

File: docusaurus/docs/Android/01-basics/01-overview.mdx [hunk 6]
```diff
@@ -29,27 +30,44 @@ The client library is a low-level wrapper around the Stream Chat API. It lets yo
 
 It also contains all the basic model objects you'll interact with when using Stream Chat, such as `User`, `Channel`, or `Message`.
 
-### Offline Support
+### State
+
+The state library is a plugin that gives you the possibility to reactively observe data through `StateFlows`.
```
Comment (MarinTolic) [inline docusaurus/docs/Android/01-basics/01-overview.mdx:35]:
```suggestion
The state library is a plugin that gives you the possibility to reactively observe data through `StateFlow`s.
```

File: docusaurus/docs/Android/01-basics/03-dependencies.mdx [hunk 7]
```diff
@@ -43,6 +43,16 @@ dependencies {
 }
 ```
 
+### State
+
+To use state library in your application, add the following dependency:
```
Comment (MarinTolic) [inline docusaurus/docs/Android/01-basics/03-dependencies.mdx:48]:
```suggestion
To use the state library in your application, add the following dependency:
```

**Review by MarinTolic — APPROVED** [review]

**Pull request merged**